### DISCUSSION TITLE
Dogfooding, or: Does OpenRewrite apply its own recipes?

### DISCUSSION THREAD

**Discussion by Bananeweizen** [openrewrite/rewrite#3298] [General] [answered]
After using several recipes successfully on some of my projects, I have been starting to create new recipes and been reading existing ones. I'm wondering if the OpenRewrite repositories actually apply their own recipes? That question came up after noticing several unused imports in rewrite-testing-frameworks, although there is a recipe to remove them.
Not sure how much value that provides from your point of view, but I generally consider projects using a dogfooding approach to have higher quality (since they can apply their own recommendations also in practice, not just in theory).

**Comment by timtebeek** [comment, marked as answer]
Hi @Bananeweizen ; We're certainly working towards that. We already have our platform at https://public.moderne.io to make it easy to run recipes at will, and we're working on some GitHub integrations to run these repeatedly and on pull requests. Just haven't gotten around to connecting the individual parts, but hope to do so soon. Indeed it's a great way to improve and harden, so it's definitely on the radar. And then hopefully you'll be able to use it just as well as we do for the same recipes against your own projects.
